### ISSUE TITLE
Cipheriv warning in Node 8

### ISSUE THREAD

**Issue by Domajno** [nlf/connect-mysql#42]
When using connect-mysql in Node 8.9.3 I see following warning in the console:
```
(node:15214) Warning: Use Cipheriv for counter mode of aes-256-ctr
```

**Comment by jas-** [comment]
Thanks for reporting this @Domajno, looks like the node.js crypto api is changing for some symmetric encryption modes. I will look at a fix soon.

**Comment by jas-** [comment]
@Domajno & @danielsharvey. A new branch is in progress to address this as well as a couple of other issues such as the incorrect encrypt then MAC, support for CCM or GCM AES cipher modes, lack of a key derivation to strengthen the supplied key helping protect the at rest data and of course the need to move away from the legacy `createCipher`.

I appreciate your patience

**Comment by jas-** [comment]
@Domajno @danielsharvey, the `master` branch has been updated to reflect changes. Could either of you confirm resolution?

**Comment by jas-** [comment]
This is addressed in `v2.2.0`, thanks for reporting this!

**Comment by danielsharvey** [comment]
Can confirm resolution and `v2.2.1` tests OK for me (noting https://github.com/nlf/connect-mysql/issues/47).
